adapters/datadog-porting-guide.md

```diff
@@ -0,0 +1,101 @@
+# Porting Guide
+
+## Purpose
+
+Aid developers who wish to port existing datadog (dd-trace-rb) instrumentation adapters to opentelemetry.
+
+## Interface
```

Comment:
This could describe subclassing OpenTelemetry::Instrumentation::Adapter and the required and optional parts of the DSL.